packages/schematics/angular/utility/workspace.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { virtualFs, workspaces } from '@angular-devkit/core';
+import { Rule, Tree } from '@angular-devkit/schematics';
+
+function createHost(tree: Tree): workspaces.WorkspaceHost {
+  return {
+    async readFile(path: string): Promise<string> {
+      const data = tree.read(path);
+      if (!data) {
+        throw new Error('File not found.');
+      }
+
+      return virtualFs.fileBufferToString(data);
+    },
+    async writeFile(path: string, data: string): Promise<void> {
+      return tree.overwrite(path, data);
+    },
+    async isDirectory(path: string): Promise<boolean> {
+      // approximate a directory check
+      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
+    },
+    async isFile(path: string): Promise<boolean> {
+      return tree.exists(path);
+    },
+  };
+}
+
+export function updateWorkspace(
+  updater: (workspace: workspaces.WorkspaceDefinition) => void | PromiseLike<void>,
+): Rule;
+export function updateWorkspace(
+  workspace: workspaces.WorkspaceDefinition,
+): Rule;
+export function updateWorkspace(
+  updaterOrWorkspace: workspaces.WorkspaceDefinition
+    | ((workspace: workspaces.WorkspaceDefinition) => void | PromiseLike<void>),
+): Rule {
```

Comment:
Nit: why do we need the extra signatures above? The arg typing here looks ok.

Reply:
cleaner in an IDE and allows for eventual separate jsdoc comments tailored to each overload.  The second form may also gain a read options parameter at a later time.